botocore/auth.py

```diff
@@ -230,7 +231,7 @@ def canonical_headers(self, headers_to_sign):
         for key in sorted_header_names:
             value = ','.join(self._header_value(v) for v in
                              sorted(headers_to_sign.get_all(key)))
-            headers.append('%s:%s' % (key, value))
+            headers.append('%s:%s' % (key, ensure_unicode(value)))
```

Comment:
Because we've got a set of header keys that are already unicode and a set of header values that are likely normal strings. When you the percent encoding it wants to decode the string using ascii, which fails. Try running this (which is run by our integration test) on python 2.6:

`'%s:%s' % (u'content-disposition', 'attachment; filename=5小時接力起跑.jpg;')`

Why don't we run into this issue now you might ask. Because sigv2 doesn't sign content disposition. It only signs `content-md5`, `content-type`, `date`, and any header that starts with `x-amz`.